### PR TITLE
1508.5 program requisition api remainder

### LINKED ISSUE #1508: Program Requisitions - ProgramSettings API
From epic: #1451 

Blocked by: https://github.com/openmsupply/open-msupply/issues/1481

Expose the following API:

```graphql
programRequisitionSettings {
   programs: ProgramRequisitionSettingNode[]
}

ProgramRequisitionSettingNode {
   # program name, i.e. HIV, TB
   name: String 
   orderTypes: ProgramOrderTypeNode[]
}

ProgramOrderTypeNode {
    # i.e. Normal/Emergency
    name: String
    id: String
    maxMOS: Number
    thresholdMOS: Number
    isEmergency: boolean
    availablePeriods: Period[]
}

PeriodNode {
    id: String
    name: String
    from: Date,
    to: Date
}
```

PeriodNode might already be implemented with https://github.com/openmsupply/open-msupply/issues/1504

### PR DESCRIPTION
**Pull request by andreievg**
Fixes #1508

last of 5 stacked PRs

# 👩🏻‍💻 What does this PR do? 

Misc changes for compilation and full test pass

* Adds records to MockData, you'll note that I didn't add a boolean and just inserted them (since there is not pre defined mock data, there is no need for the toggles)
* Add `find_many_by_program_requisition_settings_ids` and `find_many_by_program_schedule_ids`, those two were pretty simple, and i thought they didn't require separate repositories

# 🧪 How has/should this change been tested? 

* Try creating an internal order on fresh data file (with at least one store as supplier)
- [ ]  should be presented with selection of suppliers straight away
* Add a master list and a store tag for the store you are logged in to
* Add periods and period schedule
* Make master list a program list, and configure at least one order type 
* Sync and try creating remote order
- [ ] Should see that master list in program selection
- [ ] Should see there is an order type but supplier is not configured
- [ ] Should be able to see periods
*  Add this master list to supplier
- [ ] Should be able to see supplier

* Try with another program for this store
* Try with a different store and same program but different store tag and different order types and period schedule (make sure both new store and previous store still shows correct program settings after this change).

`note`, can't quite test max order per period setting yet (as you can't quite add internal order with api yet, if you want, can try setting `order type`, `program id` and `period id` manually on existing requisitions

Requisition creation

## 💌 Any notes for the reviewer?

Changes will be done here and merged to the last PR 

## Cary over

For the last `note` and existing requisitions, should they be filtered by `response` requisition ?

### CODE REVIEW

**Review by mark-prins — APPROVED** [review]
👍 ran tests, and the manual testing. all working as expected

File: server/repository/src/db_diesel/period/period_row.rs [hunk 1]
```diff
@@ -60,4 +60,14 @@ impl<'a> PeriodRowRepository<'a> {
             .optional()?;
         Ok(result)
     }
+
+    pub fn find_many_by_program_schedule_ids(
```
Comment (jmbrunskill) [inline server/repository/src/db_diesel/period/period_row.rs:64]:
```suggestion
    pub fn find_many_by_period_schedule_ids(
```
No reference to program required here right?

**Review by jmbrunskill — APPROVED** [review]
Looks good to me

File: server/repository/src/db_diesel/period/period_row.rs (hunk 1 above)
Comment (andreievg) [reply server/repository/src/db_diesel/period/period_row.rs:64]: oops, thanks

**Comment by andreievg** [comment]
> 👍 ran tests, and the manual testing. all working as expected

@mark-prins thanks very much, and thanks for your reviews

**Pull request merged**